[ML] Management: fix license unsubscribe

alvarezmelissa87 opened the pull request:
## Summary

Fixes error https://github.com/elastic/kibana/issues/59229#issuecomment-594718711

Uses `take` to limit subscription to first value. Doesn't require call to unsubscribe.
Removes 'skip' from test previously thought to be flaky

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

tsullivan commented on x-pack/legacy/plugins/ml/public/application/management/index.ts:
```diff
@@ -35,7 +35,9 @@ const licencingSubscription = plugins.licensing.license$.subscribe(license => {
     initManagementSection();
     // unsubscribe, we only want to register the plugin once.
-    licencingSubscription.unsubscribe();
+    if (licencingSubscription !== undefined) {
```
Is it a valid expectation for the `licensingSubscription` observable to end up being `undefined`? If it's undefined because of some other error higher up, we log the error where it's happening. It feels like this is a side-effect of something else, but we don't know what.

alvarezmelissa87 replied:
From looking at observable docs - looks like unsubscribing within a subscribe callback in an observable is not the recommended way of doing it. I'd imagine that it is possible since the subscription wouldn't have returned yet when this is first hit - so it could be `undefined`. 

I've confirmed that this works so it should unblock us. I'm working on a more rxjs-ey solution but didn't want to keep everyone waiting while I tested.

jgowdyelastic replied:
it's also worth noting that this is legacy code that will be rewritten soon for NP.
License checks where added here to enable ML's management plugin for the time being.

tsullivan replied:
Thanks for the confirmation that there will be future work done to improve this part of the code. I'm OK with the changes as they are to get folks unblocked.

alvarezmelissa87 commented:
Updated this as it needed an unrelated fix in master - I've updated to use `take` in the observable to remove the need to unsubscribe. Would you be up for taking another look when you get a chance? cc @tsullivan, @wylieconlon

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/)
* Commit: 91e41671f6f17d2eed9f666de6e78273ea281260

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/step_define.Transform: <DefinePivotForm /> Minimal initialization</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_transform_public_app_sections_create_transform_components_step_define/Kibana_Pipeline___x_pack_intake_agent___Transform___DefinePivotForm____Minimal_initialization)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 3 times on tracked branches: https://github.com/elastic/kibana/issues/59343


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `Transform: <DefinePivotForm /> Minimal initialization 1`

- Snapshot
+ Received

@@ -542,10 +542,11 @@
                  "attachAction": [MockFunction],
                  "clear": [MockFunction],
                  "detachAction": [MockFunction],
                  "executeTriggerActions": [MockFunction],
                  "fork": [MockFunction],
+                 "getAction": [MockFunction],
                  "getTrigger": [MockFunction],
                  "getTriggerActions": [MockFunction],
                  "getTriggerCompatibleActions": [MockFunction],
                  "registerAction": [MockFunction],
                  "registerTrigger": [MockFunction],
    at Object.test (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/step_define/step_define_form.test.tsx:40:21)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/step_define.Transform: <PivotPreview /> Minimal initialization</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_transform_public_app_sections_create_transform_components_step_define/Kibana_Pipeline___x_pack_intake_agent___Transform___PivotPreview____Minimal_initialization)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 3 times on tracked branches: https://github.com/elastic/kibana/issues/59344


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `Transform: <PivotPreview /> Minimal initialization 1`

- Snapshot
+ Received

@@ -542,10 +542,11 @@
                  "attachAction": [MockFunction],
                  "clear": [MockFunction],
                  "detachAction": [MockFunction],
                  "executeTriggerActions": [MockFunction],
                  "fork": [MockFunction],
+                 "getAction": [MockFunction],
                  "getTrigger": [MockFunction],
                  "getTriggerActions": [MockFunction],
                  "getTriggerCompatibleActions": [MockFunction],
                  "registerAction": [MockFunction],
                  "registerTrigger": [MockFunction],
    at Object.test (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/step_define/pivot_preview.test.tsx:63:21)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/source_index_preview.Transform: <SourceIndexPreview /> Minimal initialization</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_transform_public_app_sections_create_transform_components_source_index_preview/Kibana_Pipeline___x_pack_intake_agent___Transform___SourceIndexPreview____Minimal_initialization)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 3 times on tracked branches: https://github.com/elastic/kibana/issues/59345


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `Transform: <SourceIndexPreview /> Minimal initialization 1`

- Snapshot
+ Received

@@ -542,10 +542,11 @@
                  "attachAction": [MockFunction],
                  "clear": [MockFunction],
                  "detachAction": [MockFunction],
                  "executeTriggerActions": [MockFunction],
                  "fork": [MockFunction],
+                 "getAction": [MockFunction],
                  "getTrigger": [MockFunction],
                  "getTriggerActions": [MockFunction],
                  "getTriggerCompatibleActions": [MockFunction],
                  "registerAction": [MockFunction],
                  "registerTrigger": [MockFunction],
    at Object.test (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/transform/public/app/sections/create_transform/components/source_index_preview/source_index_preview.test.tsx:43:21)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
```
</details>

---
and 4 more failures, only showing the first 3.

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/","number":"30875","commit":"91e41671f6f17d2eed9f666de6e78273ea281260"}]}
PIPELINE-->

joelgriffith commented on x-pack/legacy/plugins/ml/public/application/management/index.ts:
```diff
@@ -31,11 +32,11 @@ type PluginsSetupExtended = typeof npSetup.plugins & {
 const plugins = npSetup.plugins as PluginsSetupExtended;
-const licencingSubscription = plugins.licensing.license$.subscribe(license => {
+// only need to register once
+const licensing = plugins.licensing.license$.pipe(take(1));
```
Nice, LGTM

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30932/)
* Commit: 30696a2d1bf0aa4f21bef2ec3b038b0256a02f74

### History
* :broken_heart: [Build #30875](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/) failed 91e41671f6f17d2eed9f666de6e78273ea281260

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30875/","number":"30875","commit":"91e41671f6f17d2eed9f666de6e78273ea281260"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/30932/","number":"30932","commit":"30696a2d1bf0aa4f21bef2ec3b038b0256a02f74"}]}
PIPELINE-->
